{{cookiecutter.package_name}}/src/{{cookiecutter.package_name}}/tools/make_artifacts.py

```diff
@@ -39,5 +39,154 @@ def build_artifacts(location: str, output_dir: str, append: bool, verbose: int):
         How noisy the logger should be.
 
     """
-    # Stubbed out pending further consideration of best location for this code.
-    pass
+    output_dir = Path(output_dir)
+
+    if location in project_globals.LOCATIONS:
+        path = Path(output_dir) / f'{sanitize_location(location)}.hdf'
+
+        if path.exists() and not append:
+            click.confirm(f"Existing artifact found for {location}. Do you want to delete and rebuild?",
+                          abort=True)
+            logger.info(f'Deleting artifact at {str(path)}.')
+            path.unlink()
+
+        build_single_location_artifact(path, location)
+
+    elif location == 'all':
+        # FIXME: could be more careful
+        existing_artifacts = set([item.stem for item in output_dir.iterdir()
+                                  if item.is_file() and item.suffix == '.hdf'])
+        locations = set([sanitize_location(loc) for loc in project_globals.LOCATIONS])
+        existing = locations.intersection(existing_artifacts)
+
+        if existing and not append:
+            click.confirm(f'Existing artifacts found for {existing}. Do you want to delete and rebuild?',
+                          abort=True)
+            for loc in existing:
+                path = output_dir / f'{loc}.hdf'
+                logger.info(f'Deleting artifact at {str(path)}.')
+                path.unlink()
+
+        build_all_artifacts(output_dir, verbose)
+
+    else:
+        raise ValueError(f'Location must be one of {project_globals.LOCATIONS} or the string "all". '
+                         f'You specified {location}.')
+
+
+def build_all_artifacts(output_dir: Path, verbose: int):
+    """Builds artifacts for all locations in parallel.
+
+    Parameters
+    ----------
+    output_dir
+        The directory where the artifacts will be built.
+    verbose
+        How noisy the logger should be.
+
+    Note
+    ----
+        This function should not be called directly.  It is intended to be
+        called by the :func:`build_artifacts` function located in the same
+        module.
+
+
+    """
+    from vivarium_cluster_tools.psimulate.utilities import get_drmaa
+    drmaa = get_drmaa()
+
+    jobs = {}
+    with drmaa.Session() as session:
+        for location in project_globals.LOCATIONS:
+            path = output_dir / f'{sanitize_location(location)}.hdf'
+
+            job_template = session.createJobTemplate()
+            job_template.remoteCommand = shutil.which("python")
+            job_template.args = [__file__, str(path), f'"{location}"']
```

Comment:
The double quoted location is an annoying bit about location strings when you're putting them on the command line.  We should probably make a couple of encode/decode functions for passing strings with special characters to shell commands.  If we can make sanitize_location reversible, we could just use that.  I spent a dumb amount of time trying to figure out why some of my artifact builds were failing.

Reply:
Ticket exists for making reversible "sanitize_location()"